pageserver/src/tenant.rs

```diff
@@ -5138,18 +5141,10 @@ mod tests {
             }
 
             // Perform a cycle of flush, and GC
-            let cutoff = tline.get_last_record_lsn();
-            tline
-                .update_gc_info(
-                    Vec::new(),
-                    cutoff,
-                    Duration::ZERO,
-                    &CancellationToken::new(),
-                    &ctx,
-                )
-                .await?;
             tline.freeze_and_flush().await?;
-            tline.gc().await?;
+            tenant
+                .gc_iteration(Some(tline.timeline_id), 0, Duration::ZERO, &cancel, &ctx)
+                .await?;
```

Comment:
I ignored the Slack thread about this removal of update_gc_info.

Was this removal necessary to make the test pass? If not, I'd have preferred to not remove it / remove it in a separate PR.

Anyway, the damage is done, please just mention the removal in the commit message & link to the Slack thread.

Reply:
Not required at all, I just didn't want to fix all of the callsites to do the in-place modification of `Timeline::gc_info` which this PR moves to the caller `Tenant::refresh_gc_infos_internal`.

Instead of interleaved `"refresh gc info" + "compaction" + "gc"` the tests now do the `"compaction" and "full gc"`.

Reply:
didn't notice that `timeline.` changed to `tenant.`